File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        count = 0
        ans = nums[0]
        for i in range(n):
            if ans == nums[i]:
                count += 1
            else:
                count -= 1
                if count == 0:
                    ans = nums[i]
                    count += 1
        
        return ans

File: 169-majority-element/majority-element.c
int majorityElement(int* nums, int numsSize) {
    //element that appears more than half of the size of array.
    //asume that there is always a majority element in the array
    int target = nums[0], count = 1;
    for(int i = i; i < numsSize; i++){
        if(nums[i] == target)
            count++;
        else{
            count--;
            if(count == 0){
                target = nums[i];
                count = 1;
            }
        }
    }
    return target;
}